### PR TITLE
Issue 3725: fixed the NullPointerException in the eventual assertions.

### LINKED ISSUE #3725: Sporadic test failure MetricsTest.metricsTimeBasedCacheEvictionTest
**Problem description**
We are observing `MetricsTest.metricsTimeBasedCacheEvictionTest` test failing sporadically both on Travis:
```
io.pravega.test.integration.MetricsTest > metricsTimeBasedCacheEvictionTest FAILED
    java.lang.NullPointerException
        at io.pravega.test.integration.MetricsTest.lambda$metricsTimeBasedCacheEvictionTest$1(MetricsTest.java:247)
        at io.pravega.test.common.AssertExtensions.assertEventuallyEquals(AssertExtensions.java:48)
        at io.pravega.test.integration.MetricsTest.metricsTimeBasedCacheEvictionTest(MetricsTest.java:246)
```
and Jenkins:
```
java.lang.NullPointerException
	at io.pravega.test.integration.MetricsTest.lambda$metricsTimeBasedCacheEvictionTest$0(MetricsTest.java:203)
	at io.pravega.test.common.AssertExtensions.assertEventuallyEquals(AssertExtensions.java:48)
	at io.pravega.test.integration.MetricsTest.metricsTimeBasedCacheEvictionTest(MetricsTest.java:202)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```
See the [complete log for Jenkins failure](https://github.com/pravega/pravega/files/3133206/failure_log.txt).

**Problem location**
`MetricsTest.metricsTimeBasedCacheEvictionTest`

**Suggestions for an improvement**
Fix the test.

### PR DESCRIPTION
**Pull request by kevinhan88**
**Change log description**  
Fixed the NullPointerException in the eventual assertions in the test.

**Purpose of the change**  
Fixes #3725 

**What the code does**  
Added Null Pointer test into eventual assertion to avoid NPE.

**How to verify it**  
Test should pass.

### CODE REVIEW

File: test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java [hunk 1]
```diff
@@ -187,10 +189,11 @@ public void metricsTimeBasedCacheEvictionTest() throws Exception {
             log.info("Creating Reader group : {}", readerGroupName1);
 
             readerGroupManager.createReaderGroup(readerGroupName1,
-                                                 ReaderGroupConfig.builder()
-                                                                  .stream(Stream.of(scope, STREAM_NAME))
-                                                                  .automaticCheckpointIntervalMillis(2000)
-                                                                  .build());
+                    ReaderGroupConfig
```
Comment (RaulGracia) [inline test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java]: This change seems not necessary, as the line is not exceeding the max length.

File: test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java [hunk 2]
```diff
@@ -199,18 +202,20 @@ public void metricsTimeBasedCacheEvictionTest() throws Exception {
...
 
             String readerGroupName2 = readerGroupName + "2";
             log.info("Creating Reader group : {}", readerGroupName2);
 
             readerGroupManager.createReaderGroup(readerGroupName2,
-                                                 ReaderGroupConfig.builder()
-                                                                  .stream(Stream.of(scope, STREAM_NAME))
-                                                                  .automaticCheckpointIntervalMillis(2000)
-                                                                  .build());
+                    ReaderGroupConfig.builder()
```
Comment (RaulGracia) [inline test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java]: Same here.

File: test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java (hunk 1 above)
Comment (kevinhan88) [reply test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java]: The continuation indent should be 8 only.

File: test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java (hunk 2 above)
Comment (kevinhan88) [reply test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java]: The continuation indent should be 8 only.

**Comment by RaulGracia** [comment]
Restarting the build one more time before approving.

**Pull request merged**